qa-tester/amp-qa-tester.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Plugin Name: AMP QA Tester
+ * Description: Test pre-release versions of the AMP plugin.
+ * Plugin URI: https://amp-wp.org
+ * Author: AMP Project Contributors
+ * Author URI: https://github.com/ampproject/amp-wp/graphs/contributors
+ * Version: 1.0.0
+ * Text Domain: amp-qa-tester
+ * License: GPLv2 or later
+ * Requires at least: 5.0
+ * Requires PHP: 5.6
+ *
+ * Props to @adamsilverstein (https://github.com/adamsilverstein) for laying the groundwork for this plugin.
+ *
+ * @package AmpProject\AmpWP_QA_Tester
+ */
+
+namespace AmpProject\AmpWP_QA_Tester;
+
+use WP_Error;
+
+/**
+ * Errors encountered while loading the plugin.
+ *
+ * @var WP_Error
+ */
+$_amp_qa_load_errors = new WP_Error();
+
+/**
+ * Loads the plugin.
+ *
+ * @since 1.0.0
+ */
+function load_plugin() {
+	global $_amp_qa_load_errors;
+
+	$min_php_version = '5.6';
+	$min_wp_version  = '5.0';
+
+	// If the AMP plugin is not active, we simply bail.
+	if ( ! defined( 'AMP__VERSION' ) ) {
+		return;
+	}
+
+	if ( version_compare( PHP_VERSION, $min_php_version, '<' ) ) {
+		$_amp_qa_load_errors->add(
+			'insufficient_php_version',
+			sprintf(
+				/* translators: 1: required version, 2: currently used version */
+				__( 'At least PHP version %1$s is required. Your site is currently running on PHP %2$s.', 'amp-qa-tester' ),
+				$min_php_version,
+				PHP_VERSION
+			)
+		);
+	}
+
+	if ( 'amp' === basename( dirname( __DIR__ ) ) ) {
+		$_amp_qa_load_errors->add(
+			'incorrect_plugin_location',
+			__(
+				'Please move the plugin outside of the AMP plugin directory (preferably to the the plugin directory),
+				otherwise you will risk removing this plugin when installing AMP plugin builds.',
+				'amp-qa-tester'
+			)
+		);
+	}
```

Comment:
Is this actually an issue? As far as I can tell when a PHP file is in a subdirectory, it is not included in the list of installed plugins. So perhaps this condition is not needed?

Reply:
Yes you're right. I forgot I had placed a symlink of the plugin dir in the plugins directory. Removed in f0848deda0a2310d89041030525ea32b1de2dfae.